File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector <string> morse{".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        
        unordered_set <string> transformations;
        
        for(const string& word: words){
            string result;
            for(const char c: word){
                result += morse[c - 'a'];
            }transformations.insert(result);
        }
    
        return transformations.size();
    }
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse=[".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        
        transformations=list()
        
        for word in words:
            code=''
            for i in word:
                code+=morse[ord(i)-ord('a')]
            
            if(code not in transformations):
                transformations.append(code)
                
        return len(transformations)